source/manage/code-signing-custom-builds.rst

```diff
@@ -0,0 +1,133 @@
+Code Signing Custom Builds
+==========================
+
+.. include:: ../_static/badges/allplans-selfhosted.rst
+  :start-after: :nosearch:
+
+Code signing is an essential process for ensuring the authenticity and integrity of your custom ``Mattermost`` builds. This guide provides steps on how to code sign a build using your own certificates across different operating systems: Windows, Mac, and Linux.
+
+.. important::
+
+  Code signing helps reduce security warnings from operating systems and ensures smoother, more trustworthy installations. Make sure to follow each operating system's guidelines and best practices for signing applications.
+
+Prerequisites
+-------------
+
+.. tab:: Windows
+
+   1. **Code Signing Certificate**: Obtain a certificate from a Certificate Authority (CA) or use a self-signed certificate if suitable.
+   2. **SignTool**: Available as part of the Windows SDK.
+
+.. tab:: Linux
+
+   1. **GPG Key**: Create a GPG key if you don't have one.
+   2. **GnuPG**: Install `GnuPG <https://gnupg.org/howtos/card-howto/en/ch02.html>`_ if not already installed.
+
+.. tab:: Mac
+
+   1. **Developer ID Application Certificate**: Obtain from Apple. It requires an `Apple Developer account <https://developer.apple.com/account/>`_.
+   2. **Xcode**: Ensure `Xcode <https://developer.apple.com/xcode/>`_ is installed.
+
+Process
+-------
+
+.. tab:: Windows
+
+   1. **Install SignTool**
+   
+      Install the Windows SDK to access the ``SignTool`` utility.
+
+   2. **Obtain a Code Signing Certificate**
+      
+      Purchase or create a certificate (``.pfx`` file) via a CA.
+
+   3. **Import the Certificate**
+      
+      Open the ``.pfx`` file and import it into the Windows Certificate Store.
+
+   4. **Sign the Executable**
+
+      - Open the command prompt as Administrator.
+      - Use ``SignTool`` to sign your executable:
+
+      .. code-block:: sh
+
+         signtool sign /v /s "My" /sha1 <cert hash> /fd SHA256 /tr http://timestamp.digicert.com /td SHA256 <path-to-your-executable>
```

Comment:
I think we need to step to help the user determine what the cert hash value is.

Reply:
amm, we can add one more step like

```
   4. **Find the Certificate Hash**

      To find the certificate hash, open a PowerShell window and run the following command:

      .. code-block:: sh

         Get-ChildItem -Path Cert:\CurrentUser\My | Select-Object -Property Thumbprint, Subject

      Locate your certificate in the list and copy the `Thumbprint` value, which is your certificate hash.
      ```